service/history/queues/executable.go

```diff
@@ -818,6 +823,7 @@ func NewCircuitBreakerExecutable(
 func (e *CircuitBreakerExecutable) Execute() error {
 	doneCb, err := e.cb.Allow()
 	if err != nil {
+		metrics.CircuitBreakerExecutableBlocked.With(e.metricsHandler).Record(1)
```

Comment:
Does getting an error here always indicate that the circuit is open?

Reply:
Open or semi-open. In semi-open state, it lets pass a few requests to check if the destination recovered or not.